### PR TITLE
Add linting to the design package

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Issue Number: https://github.com/graycoreio/daffodil/issues/106

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

All breaking changes in foundation-demo are fixed in this PR

### CODE REVIEW

**Comment by lderrickable** [comment]
> @lderrickable I'm not sure if it makes sense to change components like [daff-hero-title] to directives. I think we need to consider how change detection runs on directives vs. components. Could you follow-up here with some insight?

I made the changes we spoke about

**Comment by lderrickable** [comment]
I'll squish after you give the "all clear"

**Comment by xelaint** [comment]
@lderrickable do you want to change `proceed-to-checkout` and `continue-shopping` back to components in this PR?

File: apps/daffio/src/app/core/footer/footer.component.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <daff-list type="link" class="footer__links" *ngFor="let linkset of links">
       <h3 daff-list-subheader>{{ linkset.title }}</h3>
       <daff-list-item *ngFor="let link of linkset.links">
-        <a daff-link color="black" routerLink="{{ link.path }}">{{ link.title }}</a>
+        <a daff-linkcolor="black" routerLink="{{ link.path }}">{{ link.title }}</a>
```
Comment (damienwebdev) [inline apps/daffio/src/app/core/footer/footer.component.html]: I think this is a mistake.

File: apps/demo/src/app/core/footer/footer.component.html [hunk 2]
```diff
@@ -3,14 +3,14 @@
     <daff-list type="link" class="footer__links" *ngFor="let linkset of links">
       <h3 daff-list-subheader>{{ linkset.title }}</h3>
       <daff-list-item *ngFor="let link of linkset.links">
-        <a daff-link color="black" routerLink="{{ link.path }}">{{ link.title }}</a>
+        <a daff-linkcolor="black" routerLink="{{ link.path }}">{{ link.title }}</a>
```
Comment (damienwebdev) [inline apps/demo/src/app/core/footer/footer.component.html]: I think this is a mistake.

File: apps/demo/src/app/product/components/product/product.component.html [hunk 3]
```diff
@@ -27,7 +27,7 @@ <h3 daff-accordion-item-title>Details</h3>
                 <li>Praesent commodo lacus ut sapien consectetur</li>
                 <li>Integer iaculis ultrices nunc</li>
               </ul>
-              <a href="#" daff-link color="black">Learn More</a>
+              <a href="#" daff-linkcolor="black">Learn More</a>
```
Comment (damienwebdev) [inline apps/demo/src/app/product/components/product/product.component.html]: I think this is a mistake.

File: libs/design/src/atoms/form/input/input.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Input Component
 
-`daff-input` is a form control element that can be used in forms.
+`daff-input  is a form control element that can be used in forms.
```
Comment (damienwebdev) [inline libs/design/src/atoms/form/input/input.md]: Add the ` back

File: libs/design/src/atoms/link/link.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ To define a button color, add `color="[value]"` to the `<daff-link>` tag.
 
 ## Usage
 ```
-<a href="#" daff-link color="primary">
+<a href="#" daff-linkcolor="primary">
```
Comment (damienwebdev) [inline libs/design/src/atoms/link/link.md]: I think this is a mistake.

File: libs/design/src/molecules/image-gallery/image-gallery/image-gallery.component.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { Component, ViewEncapsulation } from '@angular/core';
+import { Component, ViewEncapsulation, AfterViewInit } from '@angular/core';
```
Comment (damienwebdev) [inline libs/design/src/molecules/image-gallery/image-gallery/image-gallery.component.ts]: I think this is a mistake.

File: apps/daffio/src/app/content/homepage/component/homepage.component.html [hunk 7]
```diff
@@ -43,7 +43,7 @@ <h4>Discoverable</h4>
       </daff-list>
     </div>
     <div class="pwa__link">
-      <a daff-link color="accent">Learn more about Progressive Web Apps <i class="fas fa-arrow-right"></i></a>
+      <a daff-linkcolor="accent">Learn more about Progressive Web Apps <i class="fas fa-arrow-right"></i></a>
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/daffio/src/app/content/homepage/component/homepage.component.html]: I forgot which branch this was and squashed :(. My bad. I fixed this.

File: apps/daffio/src/app/core/footer/footer.component.html (hunk 1 above)
Comment (lderrickable) [reply apps/daffio/src/app/core/footer/footer.component.html]: fixed

File: apps/demo/src/app/core/footer/footer.component.html (hunk 2 above)
Comment (lderrickable) [reply apps/demo/src/app/core/footer/footer.component.html]: fixed

File: apps/demo/src/app/product/components/product/product.component.html (hunk 3 above)
Comment (lderrickable) [reply apps/demo/src/app/product/components/product/product.component.html]: fixed

File: libs/design/src/atoms/form/input/input.md (hunk 4 above)
Comment (lderrickable) [reply libs/design/src/atoms/form/input/input.md]: fixed

File: libs/design/src/atoms/link/link.md (hunk 5 above)
Comment (lderrickable) [reply libs/design/src/atoms/link/link.md]: fixed

File: libs/design/src/molecules/image-gallery/image-gallery/image-gallery.component.ts (hunk 6 above)
Comment (lderrickable) [reply libs/design/src/molecules/image-gallery/image-gallery/image-gallery.component.ts]: fixed

**Pull request merged**